frontend/routes/components/resources/resources.component.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ *  Copyright (C) 2019 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import { RemoveIcon, IconButton,
+		ResourceItemContainer, ResourceLink, PhotoIcon, LinkIcon,
+		DocumentIcon, ResourceLabel, UploadSizeLabel, ResourcesContainer } from './resources.styles';
+
+import { LabelButton } from '../../viewer/components/labelButton/labelButton.styles';
+import AttachResourcesDialog from './attachResourcesDialog/attachResourcesDialog.container';
+import { LinearProgress } from '@material-ui/core';
+import { FieldsRow, StyledFormControl } from '../../viewer/components/risks/components/riskDetails/riskDetails.styles';
+import { FieldLabel } from '../textField/textField.styles';
+
+interface IResource {
+	_id: string;
+	name: string;
+	link: string;
+	type: string;
+	size: number;
+}
+
+interface IProps {
+	resources: IResource[];
+	onRemoveResource: (IResource) => void;
+	onSaveFiles: (files) => void;
+	onSaveLinks: (links) => void;
+	showDialog: (config: any) => void;
+}
+
+interface IState {
+	value: any;
+}
+
+export const RemoveButton = (props) => (
+	<IconButton
+		{...props}
+		aria-label="Toggle menu"
+		aria-haspopup="true"
+	>
+		<RemoveIcon />
+	</IconButton>
+);
+
+const imageExtensions = ['png', 'jpg', 'jpeg', 'gif', 'bmp', 'tiff', 'pcx'];
+
+const ResourceIcon = ({type}) =>
+	(type === 'http') ?
```

Comment:
I think it will look better if you use `cond`

Reply:
And don't need to define the same color on each variant. Instead, you can set it on `ResourceIcon`

Reply:
What do you mean with cond? conditional rendering or what?